packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+/*//////////////////////////////////////////////////////////////
+                                 IMPORTS
+    //////////////////////////////////////////////////////////////*/
+
+// Utilities
+import {Script, console2} from "forge-std/Script.sol";
+
+// Ops libs
+import {Config} from "../Config.sol";
+import {CallBatch} from "../CallBatch.sol";
+
+// Contract for selector
+import {UpgradeBeaconController} from "@nomad-xyz/contracts-core/contracts/upgrade/UpgradeBeaconController.sol";
+
+contract UpgradeCallBatches is Script, Config, CallBatch {
+    /*//////////////////////////////////////////////////////////////
+                            GOVERNANCE CALLS
+    //////////////////////////////////////////////////////////////*/
+
+    bytes upgradeHome;
+    bytes upgradeReplica;
+    bytes upgradeGovRouter;
+    bytes upgradeBridgeRouter;
+    bytes upgradeTokenRegistry;
+    bytes upgradeBridgeToken;
+
+    string configFile;
+    string[] domainNames;
+
+    function printCallBatches(
+        string memory _configFile,
+        string[] memory _domainNames,
+        string memory _localDomainName,
+        bool recovery
+    ) external {
+        localDomainName = _localDomainName;
+        domainNames = _domainNames;
+        configFile = _configFile;
+        setUp();
+        for (uint256 i; i < domainNames.length; i++) {
+            string memory domain = domainNames[i];
+            generateGovernanceCalls(domain);
+        }
+        writeCallBatch(recovery);
+    }
+
+    function setUp() internal {
+        __Config_initialize(configFile);
+        string memory outputFile = "upgradeActions.json";
+        __CallBatch_initialize(
+            localDomainName,
+            getDomainNumber(localDomainName),
+            outputFile,
+            true
+        );
+        console2.log("Governance Actions have been output to", outputFile);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                       GOVERNANCE CALL GENERATORS
+    //////////////////////////////////////////////////////////////*/
+
+    function generateGovernanceCalls(string memory domain) internal {
+        upgradeHome = abi.encodeWithSelector(
+            UpgradeBeaconController.upgrade.selector,
+            address(homeUpgrade(domain).beacon),
+            address(homeUpgrade(domain).implementation)
+        );
+        string[] memory networks = getNetworks();
```

Comment:
instead of `getNetworks`, you can use `getConnections(domain)`. then, you can remove the `if()` statement